tools/c7n_mailer/tests/test_azure.py

```diff
@@ -90,3 +92,66 @@ def test_sendgrid_handler(self, mock_send):
         result = sendgrid_delivery.sendgrid_handler(self.loaded_message, sendgrid_messages)
         self.assertTrue(result)
         mock_send.assert_called()
+
+    @patch('c7n_mailer.azure_mailer.azure_queue_processor.SmtpDelivery')
+    def test_smtp_delivery(self, mock_smtp):
+        smtp_mailer_config = {
+            'queue_url': 'asq://storageaccount.queue.core.windows.net/queuename',
+            'from_address': 'you@youremail.com',
+            'smtp_port': 25,
+            'smtp_ssl': True,
+            'smtp_server': 'test_server',
+            'smtp_username': 'user',
+            'smtp_password': 'password'
+        }
+
+        with patch('c7n_mailer.azure_mailer.sendgrid_delivery.SendGridDelivery'
```

Comment:
nit: any specific reason why we use decorator and with?

Reply:
So I can set the return value to a class variable.